### ISSUE TITLE
model preprocessing

### ISSUE THREAD

**Issue by albanie** [meetshah1995/pytorch-semseg#53]
Thanks for sharing this code.

Just a brief note that the mean subtraction used in the dataset loaders (e.g. [here](https://github.com/meetshah1995/pytorch-semseg/blob/master/ptsemseg/loader/pascal_voc_loader.py#L60)) (which was similarly used in the original FCN model [here](https://github.com/shelhamer/fcn.berkeleyvision.org/blob/master/voc-fcn8s-atonce/train.prototxt#L9)) is good for training models from scratch. However, if the segmentation model is initialized from a pytorch pretrained model (such as for the vgg16 model in fcn training code [here](https://github.com/meetshah1995/pytorch-semseg/blob/master/ptsemseg/models/__init__.py#L26)), it may be preferable to perform the preprocessing used to train that model. The pytorch [pretrained models](http://pytorch.org/docs/master/torchvision/models.html) use preprocessing that involves loading the input image in the range `[0,1]`, then performing both mean subtraction and division by rgb standard deviation:

```
from torchvision import transforms

normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                 std=[0.229, 0.224, 0.225])
```
(This wouldn't have affected the original FCN caffe code, since they use a vgg16 model that was trained with a very similar mean subtraction, but may be more significant with the different pre-processing used by the pytorch models).

**Comment by meetshah1995** [comment]
Original VGG weights expect the images to be in `img - mean`, and I'm aware that the pytorch models are different in that sense.

I guess this [line](https://github.com/meetshah1995/pytorch-semseg/blob/master/ptsemseg/loader/pascal_voc_loader.py#L98) in the code addresses the mean part. Regarding normalizing with standard deviations, it doesn't make much difference as per my experiments, even with the pytorch pre-trained models.
